src/subscription.rs

```diff
@@ -86,7 +87,14 @@ impl<'a, T: Runtime> EventSubscription<'a, T> {
             if self.finished {
                 return None
             }
-            let change_set = self.subscription.next().await;
+            let change_set = match self.subscription.next().await {
+                Some(c) => c,
+                None => {
+                    return Some(Err(
+                        RpcError::Custom("RPC subscription dropped".into()).into()
```

Comment:
this is called from `fn submit_and_watch_extrinsic` so should return an error there and be fine.

Reply:
https://github.com/paritytech/substrate-subxt/pull/214/files#diff-5cdcc3261dd77eadb37bcd60c5f85affcc9306454ae7bb8128e6d096ae51a631R561-R571